lib/workos/directory_group.rb

```diff
@@ -5,17 +5,19 @@ module WorkOS
   # The DirectoryGroup class provides a lightweight wrapper around
   # a WorkOS DirectoryGroup resource. This class is not meant to be instantiated
   # in user space, and is instantiated internally but exposed.
-  class DirectoryGroup
+  class DirectoryGroup < DeprecatedHashWrapper
     extend T::Sig
 
-    attr_accessor :id, :name
+    attr_accessor :id, :name, :custom_attributes, :raw_attributes
 
     sig { params(json: String).void }
     def initialize(json)
       raw = parse_json(json)
 
       @id = T.let(raw.id, String)
       @name = T.let(raw.name, String)
+
+      replace(to_json)
```

Comment:
@ams11 These changes are looking great! One last thing:

I noticed that just the instantiation of a `DirectoryGroup` or `DirectoryUser` will trigger a deprecation warning. I think it's because we are calling `replace`, which is a method of `Hash`, in the constructor. Is there a way we can avoid triggering the warning here?

Reply:
🤦‍♂️, doh, good call, of course it would complain about that! Let me take care of that!